File: app/src/main/java/com/khmelenko/lab/varis/log/LogEntryComposite.java
package com.khmelenko.lab.varis.log;

import java.util.ArrayList;
import java.util.List;

final class LogEntryComposite implements LogEntryComponent {

    private String mName;
    private List<LogEntryComponent> mLogEntryComponents = new ArrayList<>();

    public LogEntryComposite(String name) {
        mName = name;
    }

    void append(LogEntryComponent logEntryComponent) {
        mLogEntryComponents.add(logEntryComponent);
    }

    @Override
    public String toHtml() {
        String data = "";
        if (mName == null) {
            data += "<body style=\"background-color:#222222;\">";
        }
        for (LogEntryComponent re : mLogEntryComponents) {
            data += re.toHtml();
        }
        if (mName == null) {
            data += "</body>";
        }
        return data;
    }
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/log/LogEntryComposite.kt
package com.khmelenko.lab.varis.log

import java.util.ArrayList

internal class LogEntryComposite(val name: String?) : LogEntryComponent {
    private val logEntryComponents = ArrayList<LogEntryComponent>()

    fun append(logEntryComponent: LogEntryComponent) {
        logEntryComponents.add(logEntryComponent)
    }

    override fun toHtml(): String {
        var data = ""
        if (name == null) {
            data += "<body style=\"background-color:#222222;\">"
        }
        for (re in logEntryComponents) {
            data += re.toHtml()
        }
        if (name == null) {
            data += "</body>"
        }
        return data
    }
}
